packages/sdk/src/evm/core/classes/marketplacev3-direct-listings.ts

```diff
@@ -129,19 +142,33 @@ export class MarketplaceV3DirectListings {
   public async getAllValid(
     filter?: MarketplaceFilter,
   ): Promise<DirectListingV3[]> {
-    const startIndex = BigNumber.from(filter?.start || 0).toNumber();
     const totalListings = await this.getTotalCount();
-    const count = BigNumber.from(filter?.count || totalListings).toNumber();
 
-    if (totalListings.toNumber() === 0) {
+    let start = BigNumber.from(filter?.start || 0).toNumber();
+    let end = totalListings.toNumber();
+
+    if (end === 0) {
       throw new Error(`No listings exist on the contract.`);
     }
 
-    let rawListings =
-      await this.contractWrapper.readContract.getAllValidListings(
-        startIndex,
-        count - 1,
+    let rawListings: IDirectListings.ListingStructOutput[] = [];
+    let partialListings: any[] = [];
+    while (end - start > DEFAULT_QUERY_ALL_COUNT) {
+      partialListings.push(
+        this.contractWrapper.readContract.getAllValidListings(
+          start,
+          start + DEFAULT_QUERY_ALL_COUNT - 1,
+        ),
       );
+      start += DEFAULT_QUERY_ALL_COUNT;
+    }
+    partialListings.push(
+      await this.contractWrapper.readContract.getAllValidListings(
+        start,
+        end - 1,
+      ),
+    );
+    rawListings = (await Promise.all(partialListings)).flat();
```

Comment:
this code is duplicated in a few places - can we make a helper function for it? there's already a common/marketplace.ts file where you can put this